tokio/src/io/interest.rs

```diff
@@ -116,12 +164,51 @@ impl Interest {
     /// assert!(BOTH.is_readable());
     /// assert!(BOTH.is_writable());
     pub const fn add(self, other: Interest) -> Interest {
-        Interest(self.0.add(other.0))
+        Self(self.0 | other.0)
     }
 
     // This function must be crate-private to avoid exposing a `mio` dependency.
-    pub(crate) const fn to_mio(self) -> mio::Interest {
-        self.0
+    pub(crate) fn to_mio(self) -> mio::Interest {
+        // mio does not allow and empty interest, so pick an arbitrary starting value
+        let mut mio = mio::Interest::READABLE;
```

Comment:
```suggestion
    pub(crate) fn to_mio(self) -> mio::Interest {
    	fn mio_add(wrapped: &mut Option<mio::Interest>, add: mio::Interest) {
	    match wrapped {
	        Some(inner) => *inner |= add,
		None => *wrapped = Some(add),
	    }
	}

        // mio does not allow and empty interest, so use None for empty
        let mut mio = None;
```

Reply:
an option is clearer, but needs to be unwrapped. I've documented it heavily

Reply:
You can just use `mio.unwrap_or(mio::Readiness::READABLE)`.